app/stream/index/chat/controller.js

```diff
@@ -55,6 +58,14 @@ export default Controller.extend({
 
   },
 
+  commentsSectionScroll() {
+    this.timer = run.debounce(this, function() {
+      if (this.$commentsElement.scrollTop() < 10) {
+        this.send('loadEarlier');
+      }
+    }, 20000, true);
+  },
+
   keyboardPusherOptions: {
```

Comment:
Looks good but we shouldn't attempt to fetch earlier comments if they're already all displays. You can use the isShowingAllComments property to check this